src/layers.h

```diff
@@ -11,6 +11,10 @@
   const Key keymaps[][ROWS][COLS] PROGMEM = { layers };		\
   uint8_t layer_count = sizeof(keymaps) / sizeof(*keymaps);
 
+// The maximum number of layers allowed. `LayerState`, which stores
+// the on/off status of the layers in a bitfield has only 32 bits, and
+// that should be enough for almost any layout.
+#define MAX_LAYERS 32
```

Comment:
Is there a reason why you ended up not defining MAX_LAYERS in the way you mentioned earlier, where it's clear how it's derived?

Reply:
I thought it might be useful outside of `layers.cpp`, where `LayerState` is declared. On second thought, it's probably better right next to the `LayerState` declaration, and not really relevant anywhere else.

Reply:
Oh. Sorry, I didn't mean "where it's defined" but "how it's defined" You'd had the calculation in terms of the sizeof(LayerState), which made a lot more sense than just the 32. If we _can_ get the compiler to figure out that it's 32, that's better than us hardcoding it.

Reply:
Yes, I understood that; it's just that I couldn't define it that way in a file where `LayerState` wasn't declared. I moved it to `layers.cpp` in another commit, and did define it that way.